milestones/api.py

```diff
@@ -77,12 +77,19 @@ def get_milestone_relationship_types():
     return data.fetch_milestone_relationship_types()
 
 
-def add_milestone(milestone):
+def add_milestone(milestone, propagate=True):
     """
     Passes a new milestone to the data layer for storage
+
+    Arguments:
+        milestone (dict): The milestone to persist
+        propagate (bool): False to prevent reactivation of soft-deleted milestone relationships
```

Comment:
The gating use case here is as follows:

1. Make a subsection a prerequisite (create Milestone, create 'fulfills' CourseContentMilestone)
2. Gate another subsection using this prerequisite subsection (create 'requires' CourseContentMilestone)
3. Unmake the prerequisite subsection (remove Milestone, which also removes the two CourseContentMilestones)
4. Remake the subsection a prerequisite (At this point we don't won't to automatically restore the 'requires' CourseContentMilestone relationship created above)